src/custom/hooks/useFetchFile.ts

```diff
@@ -1,15 +1,23 @@
 import { useEffect, useState } from 'react'
 
+function getErrorMessage(filePath: string, res: Response): string {
+  return `Error fetching file ${filePath} - status: ${res.statusText}`
+}
+
 export default function useFetchFile(filePath: string) {
   const [file, setFile] = useState<string | null>(null)
   const [error, setError] = useState<string | null>(null)
 
   useEffect(() => {
     const fetchFile = async () => {
       await fetch(filePath)
-        .then((res) => res.text())
-        .then((text) => {
-          setFile(text)
+        .then(async (res) => {
+          if (res.ok) {
+            const fileContent = await res.text()
+            setFile(fileContent)
+          } else {
+            setError(getErrorMessage(filePath, res))
+          }
```

Comment:
this is a bonus, it was not handling the not ok case. I was getting 404 as a success